sechub-pds/src/main/java/com/mercedesbenz/sechub/pds/execution/PDSExecutionEnvironmentService.java

```diff
@@ -81,55 +80,57 @@ private void calculateAndSetupEnvironment(UUID pdsJobUUID, PDSJobConfiguration c
         LOG.debug("Initialized environment variables for script of pds job: {} with: {}", pdsJobUUID, environment);
     }
 
-    private Map<String, String> initCleanEnvironment(PDSProductSetup productSetup, ProcessBuilder builder) {
+    private PDSSafeProcessEnvironmentAccess initCleanEnvironment(PDSProductSetup productSetup, ProcessBuilder builder) {
         Map<String, String> environment = builder.environment();
 
         Set<String> pdsScriptEnvWhitelist = productSetup.getEnvWhitelist();
         LOG.debug("PDS script environment variable white list: '{}'", pdsScriptEnvWhitelist);
 
         environmentCleaner.clean(environment, pdsScriptEnvWhitelist);
 
-        return environment;
+        PDSSafeProcessEnvironmentAccess result = new PDSSafeProcessEnvironmentAccess(builder.environment());
```

Comment:
I think the environment should be hand over here? - otherwise the environmentCleaner has no effect.
```suggestion
        PDSSafeProcessEnvironmentAccess result = new PDSSafeProcessEnvironmentAccess(environment);
```

Reply:
Thank you for finding this! You are absolutely right - this was wrong!